fdbclient/ManagementAPI.actor.cpp

```diff
@@ -853,8 +855,18 @@ ACTOR Future<CoordinatorsResult::Type> changeQuorum( Database cx, Reference<IQuo
 				return CoordinatorsResult::BAD_DATABASE_STATE;  // Someone changed the "name" of the database??
 
 			state CoordinatorsResult::Type result = CoordinatorsResult::SUCCESS;
-			std::vector<NetworkAddress> _desiredCoordinators = wait( change->getDesiredCoordinators( &tr, old.coordinators(), Reference<ClusterConnectionFile>(new ClusterConnectionFile(old)), result ) );
-			std::vector<NetworkAddress> desiredCoordinators = _desiredCoordinators;
+			if(!desiredCoordinators.size()) {
+				std::vector<NetworkAddress> _desiredCoordinators = wait( change->getDesiredCoordinators( &tr, old.coordinators(), Reference<ClusterConnectionFile>(new ClusterConnectionFile(old)), result ) );
+				desiredCoordinators = _desiredCoordinators;
+			}
+
+			if(result == CoordinatorsResult::NOT_ENOUGH_MACHINES && notEnoughMachineResults < 1) {
+				//we could get not_enough_machines if we happen to see then database while the coordinator is updating the worker list, so make sure it happens twice before returning a failure
```

Comment:
```suggestion
				//we could get not_enough_machines if we happen to see the database while the coordinator is updating the worker list, so make sure it happens twice before returning a failure
```

Also, is this while the coordinator is updating the worker list, or is it the cluster controller?

Reply:
you are right, its the cluster controller